### ISSUE TITLE
Table:  Investigate bring in reactabular-resizable

### ISSUE THREAD

**Issue by tlabaj** [patternfly/patternfly-react#7067]
**Describe the issue. What is the expected and unexpected behavior?**
A request was made for support of resizable columns.  There is an extension for `reactabular` called `reactabular-resizable`.   This spike is to investigate bring this into the PF react-table component.

**Comment by jenny-s51** [comment]
After making a codesandbox demo: https://codesandbox.io/s/reactabular-demo-forked-ics44n?file=/src/App.tsx, I did some investigating as to how we could potentially integrate reactabular-resizable with PF, and reached out to @jeff-phillips-18 who provided some great insight regarding this issue.

We found that our PF implementation of the legacy table is built over the reactabular table, therefore bringing in reactabular-resizable for our legacy table would be feasible. 

However, for the composable table, we would need to either create our own in-house solution to resizing columns, or use a different library that supports our current implementation of composable table. 

@tlabaj what are your thoughts on how we should proceed here?

Do we want to use reactabular-resizable for the legacy table, and investigate a different solution for composable table? If that is the case, we would need a design for a variant with resizable columns and styling of the draggable handle.
